### ISSUE TITLE
Desktop switcher in lxqt-panel won't refresh desktop name after changing in obconf

### ISSUE THREAD

**Issue by oouuppss** [lxde/lxqt#668]
I'am unable to set some desktop name on lxqt-panel "Changeur de bureau" widget  using obconf
Archlinux 4.0.4-2-ARCH   x86_64
obconf 2.0.4-2
lxqt-panel 0.9.0-1
liblxqt 0.9.0-2

**Comment by npmiller** [comment]
> lxqt-panel 0.9.0-1
> liblxqt 0.9.0-2

Are these the packages in the `community` repository ? If yes, you should try with the `*-git` packages from the aur as the 0.9 release is getting a bit old and a lot of changes have been made since (including big changes to the desktop switch).

**Comment by pmattern** [comment]
Indeed. In recent VCS versions the name of virtual desktops can be changed from the configuration dialog of 'Desktop switcher' (='Changeur de bureau' en français), so switching to VCS packages from AUR seems to be the only option to get this working right now.

But this doesn't mean this issue is solved IMO:
In recent VCS versions of ObConf-Qt the pane used to set desktop names seems to be way blocked but still exists. Similar situation regarding e. g. xfwm4-workspace-settings, the counterpart of Xfwm4, where the corresponding list gets blocked as well if the tool gets launched in an LXQt session.
Nevertheless this means one distinct feature can be configured at two different locations and only one of those works. This doesn't make for a good user experience and should be changed somehow.

**Comment by oouuppss** [comment]
> npmiller
> lxqt-panel 0.9.0-1
> liblxqt 0.9.0-2
> Are these the packages in the community repository ? 
Yes they are, ok.

agree with pmattern ;)

**Comment by paulolieuthier** [comment]
The desktop switch plugin uses KWindowSystem as its backend, so it should work. If it's not working, then we probably should remove that feature and let the window manager handle.

**Comment by npmiller** [comment]
It does work. There are multiple problems though.

First it's only temporary, as it's not stored in lxqt config files, and at least openbox doesn't update its configuration accordingly, so if you login/out the changes you made at the desktop switch level will be gone.

Also the window manager might not expect anything else changing these, for example obconf preloads the desktop names from the config file and not from the *actual* names so if you change the names from the desktop switch then launch obconf you will have a mismatch.

Overall, to make it work properly we would *need* some window manager specific code which I don't think is a good idea, so I agree that we should remove this feature and let the window manager handle it.

**Comment by wxl** [comment]
I had never noticed this before, but it does seem that there is a big problem here even outside of obconf. As @npmiller said, the desktop name is not stored in any LXQt config file. Using the default mechanism to change the desktop name (i.e. through lxqt-panel itself) thus has the result of being only a temporary change. I've tried this with KWin and it's the same problem, so it's not just Openbox that's the issue here.

There's a [downstream bug](https://bugs.launchpad.net/ubuntu/+source/lxqt-panel/+bug/1849877) that drew my attention to this and thus here.

**Comment by tsujan** [comment]
I saw it years ago but forgot it because I don't use the switcher (I'm addicted to easystroke's mouse gestures and libinput-gestures' touchpad gestures).

I don't think we can change desktop names permanently; the window manager should do it. If so, this is a confusing feature and should be removed.

**Comment by tsujan** [comment]
And please note that, in 2015, users might have thought openbox had a special relation to LXQt. If the switcher can really handle desktop names, it should do it with all WMs, which seems impossible to me (but I'm not sure yet).

**Comment by wxl** [comment]
Right. So let's tear it out of there and close this up :)

**Comment by tsujan** [comment]
First, someone should take a look at the code. You might use swear words if you know that "plugin-desktopswitch" is the only folder I've never opened when working on the panel ;) But I'll do it later.

**Comment by stefonarch** [comment]
I use the switcher and like to see names (related to tasks) and the not numbers.

The switcher overtakes the names when set  in obconf with openbox or systemsettings5 for kwin, so it's enough to take out only  the name setting part.

**Comment by Sinihopea** [comment]
Hello, I just came to message that I actually use this feature and it works as intended at least with WindowMaker. Please, don't rip it out just yet. If I can help you with the code so that it works with other window managers then I offer my help. Although I don't know much about Qt or X11 code but I might be able to put together something which works independently of window managers as long as the plugin-desktopswitch gets number of virtual desktops from somewhere.

**Comment by stefonarch** [comment]
Afaik it works also for Kwin(_wayland). So we could add a tooltip only, saying that it may not be persistent on some window managers.
